include/range/v3/view/subrange.hpp

```diff
@@ -0,0 +1,428 @@
+              : data_{detail::move(i), detail::move(s), meta::nil_{}}
+            {}
+
+            CONCEPT_REQUIRES(StoreSize)
+            constexpr subrange(I i, S s, difference_type_t<I> n)
+              : data_{(RANGES_ASSERT(!RandomAccessIterator<I>() ||
+                    ranges::next(i, n) == s), detail::move(i)),
+                    detail::move(s), n}
+            {}
+
+            CONCEPT_REQUIRES(SizedSentinel<S, I>())
+            constexpr subrange(I i, S s, difference_type_t<I> n)
+              : data_{(RANGES_ASSERT(s - i == n), detail::move(i)),
+                    detail::move(s), ((void)n, meta::nil_{})}
+            {}
+
+            template<typename R,
+                CONCEPT_REQUIRES_(!StoreSize &&
+                    ReferenceableRange<R>() &&
+                    ConvertibleTo<iterator_t<R>, I>() &&
+                    ConvertibleTo<sentinel_t<R>, S>())>
+            constexpr subrange(R &&r)
+              : subrange{ranges::begin(r), ranges::end(r)}
+            {}
+
+            template<typename R,
+                CONCEPT_REQUIRES_(StoreSize &&
+                    ReferenceableRange<R>() &&
+                    SizedRange<R>() &&
+                    ConvertibleTo<iterator_t<R>, I>() &&
+                    ConvertibleTo<sentinel_t<R>, S>())>
+            constexpr subrange(R &&r)
+              : subrange{ranges::begin(r), ranges::end(r), ranges::distance(r)}
+            {}
+
+            template<typename R,
+                CONCEPT_REQUIRES_(K == subrange_kind::sized &&
+                    ReferenceableRange<R>() &&
+                    ConvertibleTo<iterator_t<R>, I>() &&
+                    ConvertibleTo<sentinel_t<R>, S>())>
+            constexpr subrange(R &&r, difference_type_t<I> n)
+              : subrange{ranges::begin(r), ranges::end(r),
+                    (RANGES_ASSERT(!SizedRange<R>() || n == ranges::distance(r)), n)}
+            {}
+
+            template<class PairLike,
+                CONCEPT_REQUIRES_(!StoreSize &&
+                    detail::PairLikeConvertibleTo<PairLike, I, S>())>
+            constexpr subrange(PairLike &&r)
+            : subrange{std::get<0>(static_cast<PairLike &&>(r)),
+                std::get<1>(static_cast<PairLike &&>(r))}
+            {}
+
+            template<class PairLike,
+                CONCEPT_REQUIRES_(K == subrange_kind::sized &&
+                    detail::PairLikeConvertibleTo<PairLike, I, S>())>
+            constexpr subrange(PairLike &&r, difference_type_t<I> n)
+            : subrange{std::get<0>(static_cast<PairLike &&>(r)),
+                std::get<1>(static_cast<PairLike &&>(r)), n}
+            {}
+
+            template<typename PairLike,
+                CONCEPT_REQUIRES_(!Range<PairLike>() &&
+                    detail::PairLikeConvertibleFrom<PairLike, I const &, S const &>())>
+            constexpr operator PairLike() const
+            {
+                return PairLike(first_(), last_());
+            }
```

Comment:
I'm supposing this is enough to make `subrange` convertible to `iterator_range`, correct?